integration-core/src/main/java/org/symphonyoss/integration/core/bridge/IntegrationBridgeImpl.java

```diff
@@ -65,24 +66,44 @@
     return sendMessage(instance, integrationUser, streams, message);
   }
 
+  /**
+   * It receives an array of Stream and sends one by one to each room.
+   * Business rule 1: If I have "n" rooms and one receive OK then returns HTTP Status 200
+   * Business rule 2: If I have "n" rooms and all returned failed the priority is to return HTTP Status 500
+   * @param instance the instance of integration
+   * @param integrationUser the user of integration
+   * @param streams the array of stream.
+   * @param message the actual message. It's expected to be already on proper format.
+   * @return
+   * @throws RemoteApiException
+   */
   @Override
   public List<Message> sendMessage(IntegrationInstance instance, String integrationUser,
-      List<String> streams, Message message) {
+      List<String> streams, Message message) throws RemoteApiException {
     List<Message> result = new ArrayList<>();
 
+    RemoteApiException remoteApiException = null;
     for (String stream : streams) {
       try {
         Message messageResponse = postMessage(integrationUser, stream, message);
         result.add(messageResponse);
       } catch (RemoteApiException e) {
         exceptionHandler.handleRemoteApiException(e, instance, integrationUser, stream);
+        if(remoteApiException == null || Response.Status.fromStatusCode(e.getCode()).getFamily() == Response.Status.Family.SERVER_ERROR) {
```

Comment:
SERVER_ERROR only includes 5xx errors.  What about 4xx errors?  Might be better to check for != SUCCESS.